HDDS-7612. Add documentation for container balancer

david1859168 opened the pull request:
## What changes were proposed in this pull request?
Added the content about "Container balancer" under the "feature" menu from "https://docs.cloudera.com/cdp-private-cloud-base/7.1.8/ozone-storing-data/topics/ozone-container-balancer-cli.html"

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7612

Galsza commented:
Thank you for working on this @david1859168 . Right now the documentation is also missing an overall description about what the container balancer is. I'd prefer if you could also add information on it from [container balancer overview](https://docs.cloudera.com/cdp-private-cloud-base/7.1.8/ozone-storing-data/topics/ozone-container-balancer-overview.html )

david1859168 commented:
Hi @Galsza I have added the [container balancer overview](https://docs.cloudera.com/cdp-private-cloud-base/7.1.8/ozone-storing-data/topics/ozone-container-balancer-overview.html), please review.

Galsza commented:
@david1859168 Hi, thank you for updating the documentation. Official Apache documentation shouldn't reference Cloudera or Cloudera specific items so please remove those.

Galsza commented on hadoop-hdds/docs/content/feature/ContainerBalancer-cfg.md:
```diff
@@ -0,0 +1,41 @@
+menu:
+   main:
+      parent: "Container Balancer overview"
+summary: To use container balancer using Cloudera Manager, perform the following steps.
```
This file shouldn't be used as it only provides Cloudera specific steps and information

Galsza commented on hadoop-hdds/docs/content/feature/ContainerBalancer-cm.md:
```diff
@@ -0,0 +1,40 @@
+menu:
+   main:
+      parent: "Container Balancer overview"
+summary: You can activate and deactivate container balancer feature using Cloudera Manager.
```
This file is not needed either as it also directly about Cloudera related steps

Galsza commented on hadoop-hdds/docs/content/feature/ContainerBalancerOverview.md:
```diff
@@ -0,0 +1,41 @@
+  limitations under the License.
+-->
+
+## Container Balancer overview
```
Yes, my original comment about feature overview was referring to this. Based on this the user gets an idea about what the container balancer is and what it does. Thank you for adding it.

Galsza commented on hadoop-hdds/docs/content/feature/ContainerBalancerOverview.md:
````diff
@@ -0,0 +1,41 @@
+```
+## Container balancer CLI commands({{< ref "feature/ContainerBalancer-cli.md" >}})
+
+## Configuring container balancer service({{< ref "feature/ContainerBalancer-cfg.md" >}})
````
Please remove these references as well when the Cloudera specific items no longer exist

Galsza commented on hadoop-hdds/docs/content/feature/ozone-container-balancer1.png:
Please remove this picture as it is Cloudera specific.

Galsza commented on hadoop-hdds/docs/content/feature/ozone-container-balancer2.png:
Please remove this picture as it is Cloudera specific

Galsza reviewed (changes requested):
I've marked the places where I see Cloudera specific documentation, please avoid using it as Apache Ozone is open source and shouldn't be tied to a specific Company.

david1859168 commented:
Thank you @Galsza. I have modified and deleted the CM-related documents, please verify. 	
        deleted:    ContainerBalancer-cfg.md
	deleted:    ContainerBalancer-cm.md
	modified:   ContainerBalancerOverview.md
	deleted:    ozone-container-balancer1.png
	deleted:    ozone-container-balancer2.png

Galsza commented:
@david1859168 thank you for you update, LGTM +1

@errose28 could you please take a look at this?

errose28 reviewed (changes requested):
@david1859168 we can't commit this in its current state. Thereare numerous problems that must be addressed:

1. The resulting format is not as expected. Please use `hugo serve` from the `ozone/hadoop-hdds/docs` to see what the generated documentation looks like.
2. Wording cannot be copied directly from other sources. Please explain the container balancer in your own words.
3. Content is overall lacking. We need more explanation of what the container balancer is, how it works, what CLIs are available and when to use them. For example, there are numerous configurations about what constitutes "unbalanced" and how much data to move on each iteration.

@siddhantsangwan can you help @david1859168 make more thorough docs for the balancer? It seems documentation updates were missed as part of that feature.

adoroszlai commented:
> 1. Please use `hugo serve` from the `ozone/hadoop-hdds/docs` to see what the generated documentation looks like.

Ozone build installs Hugo if not foud locally, so you can also use:

```
mvn -pl :hdds-docs clean package
open hadoop-hdds/docs/target/classes/docs/index.html
```

Afterwards, Hugo can be found at `.dev-tools/hugo/bin/hugo`

david1859168 commented:
> @david1859168 we can't commit this in its current state. Thereare numerous problems that must be addressed:
> 
> 1. The resulting format is not as expected. Please use `hugo serve` from the `ozone/hadoop-hdds/docs` to see what the generated documentation looks like.
> 2. Wording cannot be copied directly from other sources. Please explain the container balancer in your own words.
> 3. Content is overall lacking. We need more explanation of what the container balancer is, how it works, what CLIs are available and when to use them. For example, there are numerous configurations about what constitutes "unbalanced" and how much data to move on each iteration.
> 
> @siddhantsangwan can you help @david1859168 make more thorough docs for the balancer? It seems documentation updates were missed as part of that feature.

Hi @siddhantsangwan, there are a few pdf and Google docs available at https://issues.apache.org/jira/browse/HDDS-4656.  Which one do you think should be put here?

adoroszlai commented:
@siddhantsangwan can you please revisit this?

adoroszlai commented:
Thanks @david1859168 for the contribution.  We will need to revisit this.
